opba-db/src/main/java/de/adorsys/opba/db/repository/BankRepositoryImpl.java

```diff
@@ -3,27 +3,38 @@
 import de.adorsys.opba.db.domain.entity.Bank;
 import lombok.RequiredArgsConstructor;
 import org.springframework.jdbc.core.BeanPropertyRowMapper;
-import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 @Repository
 @RequiredArgsConstructor
 public class BankRepositoryImpl {
 
     private static final BeanPropertyRowMapper<Bank> ROW_MAPPER = BeanPropertyRowMapper.newInstance(Bank.class);
 
-    private final JdbcTemplate jdbcTemplate;
+    private final NamedParameterJdbcTemplate jdbcTemplate;
 
     public List<Bank> getBanks(String query, int startPos, int maxResults) {
-        String sql = "SELECT id, uuid, name, bic, bank_code, "
-                + "word_similarity(?, (name || ' ' || bic || ' ' || bank_code)) as sim "
-                + "FROM opb_bank "
-                + "WHERE ? <% (name || ' ' || bic || ' ' || bank_code)"
-                + "ORDER BY sim DESC "
-                + "LIMIT ? "
-                + "OFFSET ?";
-        return jdbcTemplate.query(sql, ROW_MAPPER, query, query, maxResults, startPos);
+        String subQuery = "SELECT id, uuid, name, bic, bank_code, word_similarity(:keyword, field) as sml "
+                + "FROM opb_bank WHERE :keyword <% field ";
+
+        String sql = Stream.of("name", "bank_code", "bic")
+                .map(s -> subQuery.replaceAll("field", s))
+                .collect(Collectors.joining("UNION "))
+                + "ORDER BY sml DESC "
```

Comment:
use `ORDER BY sml DESC, id` to ensure consistent order for words having same `sml`

Reply:
done